src/exchange/CeloNewsFeedItem.tsx

```diff
@@ -0,0 +1,77 @@
+import { formatDistanceToNowStrict } from 'date-fns'
+import * as React from 'react'
+import { Image, StyleSheet, Text, View } from 'react-native'
+import Touchable from 'src/components/Touchable'
+import { CeloNewsArticle } from 'src/exchange/types'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+const IMAGE_SIZE = 44
+
+interface Props {
+  article: CeloNewsArticle
+  testID?: string
+}
+
+export default function CeloNewsFeedItem({ article, testID }: Props) {
+  function onPress() {
+    navigate(Screens.WebViewScreen, { uri: article.link })
+  }
```

Comment:
```suggestion
  useCallback(function onPress() {
    navigate(Screens.WebViewScreen, { uri: article.link })
  }, [article.link])
```

Reply:
There are some handy linting rules that should help us catch un-memoized props, I'll look at adding them to the wallet.

Reply:
this is my opinion but i don't believe that pre or over optimising brings the best outcomes...there's an upfront cost to memoising something, are we really going to gain much from memoising simple functions like these here?

Reply:
+1 to what @kathaypacific said. My memory is a bit fuzzy but I think we discussed this in the team a long time ago and decided to not only memoize callbacks when we identify clear bottlenecks.
I'll try to find the discussion.

Reply:
To be clear - this is to prevent re-renders of children components due to referential inequality, not because the function itself is expensive.

I think there are generally two schools of thought regarding memoization.

1) **Memoize when it's needed** - When performance issues arise (due to either intense computations while rendering, or massive re-renders because of unstable references as props), you add memoization. This requires that you monitor for performance issues. It also means breaking out some tooling and combing through component trees to identify the source of performance issues. Fortunately React is generally really fast. Unfortunately, if you _do_ find yourself in this position, fixing the issue is generally non-trivial - but that's just my experience. There are wizards in this realm that can identify and fix perf issues very quickly.

2) **Memoize aggressively to pre-empt performance issues** - Memoize all non-primitive values passed as props to child components. Yes, this is a pre-optimization. The cost is more memory usage (although I believe it's really quite a small increase, as React stores props between renders regardless if the value is memoized) and cognitive load. Fortunately the cognitive load can be solved with linting. The benefit here is that you're much less likely to need to invest significant time tracking down performance issues.

I wouldn't say I have a _strong_ opinion in this regard, but I've generally been a member of the second camp. I'd rather invest a little up-front cost in adding linting rules to enforce memoization, than potentially have to invest a bunch of time tracking down and fixing performance issues in the future.

Reply:
I'm ok with that approach if it's something the team has intentionally decided upon. I'd love to find the discussion if you can find it.

Reply:
The code legibility hit if we use memoization for everything is also a consideration. It would be interesting to see the results of the profiler for this simple case to see if there is enough of an improvement to justify using memoize more aggressively. Right now I don't think we a clear picture on how much of a performance boost we would get with memoization, but think that would be helpful for the discussion.

Reply:
maybe it is worth discussing again but i am firmly in the first group you describe @jacob-israel-turner. performance optimisations often comes at a cost (legibility and maintenance, as Tom mentions above) and in my opinion we should be intentional and strategic about what we are doing. not only in the performance topic but also more generally. this component is already a leaf component so i'm not sure that i understand your point about preventing re-renders of children components here, it seems very inexpensive to let react native handle re-renders of views and texts and images efficiently.